src/project.rs

```diff
@@ -284,15 +314,8 @@ impl FuzzProject {
             cmd.arg("--bins");
         }
 
-        if let Some(target_dir) = &build.target_dir {
-            cmd.arg("--target-dir").arg(target_dir);
-        } else if build.coverage {
-            // To ensure that fuzzing and coverage-output generation can run in parallel, we
-            // produce a separate binary for the coverage command.
-            let target_dir = env::current_dir()?
-                .join("target")
-                .join(default_target())
-                .join("coverage");
+        if build.target_dir.is_some() || build.coverage {
+            let target_dir = self.target_dir(&build)?;
```

Comment:
Can we make the `target_dir` method return an `Option` and do `if let Some(target) = self.target_dir(&build)` here? That way we are less likely to specify the `--target-dir` flag to `cargo` when we intend to use the default (and the default might have been configured to a non-default location by the user in `.cargo/config.toml` or via env vars so it is best to omit the `--target-dir` flag than to guess its location by appending `<project>/target` or something like that)